python/cuml/common/input_utils.py

```diff
@@ -55,6 +55,51 @@ def get_cudf_column_ptr(col):
     return col.__cuda_array_interface__['data'][0]
 
 
+def get_supported_input_type(X):
+    """
+    Determines if the input object is a supported input array-like object or
+    not. If supported, the type is returned. Otherwise, `None` is returned.
+
+    Parameters
+    ----------
+    X : object
+        Input object to test
+
+    Returns
+    -------
+    array-like type or None
+        If the array-like object is supported, the type is returned.
+        Otherwise, `None` is returned.
+    """
+    if (isinstance(X, cudf.Series)):
+        if X.null_count != 0:
+            return None
+        else:
+            return cudf.Series
+
+    # converting pandas to numpy before sending it to CumlArray
+    if isinstance(X, pd.DataFrame):
+        return pd.DataFrame
+
+    if isinstance(X, pd.Series):
+        return pd.Series
+
+    if isinstance(X, cudf.DataFrame):
+        return cudf.DataFrame
+
+    if isinstance(X, CumlArray):
+        return CumlArray
+
+    if hasattr(X, "__cuda_array_interface__"):
+        return cp.ndarray
```

Comment:
Good catch. I adapted this from `input_to_cuml_array` which will treat these objects as a `cupy.ndarray` if those interfaces are found and incorrectly assumed that meant they are that type. But now I am not sure what the correct return value should be.

The goal of the method is to determine if 1) this object can be converted with `input_to_cuml_array`, and if it can be, then 2) what type it is. For my purposes I really just need whether it is one of the available types accepted by `to_output()` (i.e. "numpy", "cupy", "cudf", etc).

The easiest solution would probably be to convert it to a boolean return and then do individual testing as needed. Thoughts?

Reply:
I think leaving the method as is and just changing the docstring to reflect the purpose as you mentioned it would be perfect